src/common/NameSpaceUtil.cpp

```diff
@@ -28,7 +28,7 @@ bool NameSpaceUtil::isEndPointURL(string nameServerAddr) {
 }
 
 string NameSpaceUtil::formatNameServerURL(string nameServerAddr) {
-  int index = nameServerAddr.find(ENDPOINT_PREFIX);
+  unsigned int index = nameServerAddr.find(ENDPOINT_PREFIX);
```

Comment:
BUG !!!
when there is no ENDPOINT_PREFIX in nameServerAddr, index will equal to 0xffffffff,
that means never equal to string::nps.

Reply:
@ShannonDing

Reply:
https://github.com/apache/rocketmq-client-cpp/pull/147

I made a PR for this bug.